Develop3

SergeyDyomin92 opened the pull request:
Написаны UI-автотесты для проекта StellarBergers.
Использован паттерн PageObject. Покрыты тестами страницы:

- Домашняя страница
- Хедер
- Страница регистрации
- Страница входа
- Страница восстановления пароля
- Страница "Личный кабинет"

Перед каждым тестом создавался новый пользователь по API методам.
После каждого теста сгенерированный пользователь удалялся по API.
Автотесты проверены в двух браузерах с использованием паттерна Factory:
- Google Chrome
- Yandex Browser 

Создан Allure-отчёт по всем проверкам. Дефектов в реализации не обнаружено.

Используемые библиотеки и плагины:
- selenium-java 3.141.59
- junit 4.13.2
- aspectj 1.9.7
- allure 2.15.0
- rest-assured 5.4.0
- gson 2.8.5
- java-faker 1.0.2
- maven-surefire-plugin 2.22.2
- allure-maven 2.10.0 (reportVersion 2.15.0)

SergeyDyomin92 committed:
Created UI-autotests with Selenium and AllureReport

SergeyDyomin92 committed:
Created UI-autotests with Selenium and AllureReport (небольшие правки)

nikolay-977 commented on src/test/java/site/nomoreparties/stellarburgers/ConstructorTest.java:
```diff
@@ -0,0 +1,156 @@
+    private static final String URL = Url.URL;
+
+    private WebDriver driver;
+    String name;
```
Необходимо исправить: Для всех полей следует указывать минимально необходимые модификаторы доступа

nikolay-977 commented on src/test/java/site/nomoreparties/stellarburgers/ConstructorTest.java:
```diff
@@ -0,0 +1,156 @@
+
+import static site.nomoreparties.stellarburgers.driver.WebDriverCreator.createWebDriver;
+
+public class ConstructorTest {
```
Можно улучшить: применить наследование - создать класс BaseTest и вынести в него методы с аннотациями `@Before` и `@After`, где это возможно. Это позволит избежать повторяющегося кода и сократит код автотестов.

nikolay-977 reviewed (commented):
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и как запускать (`mvn clean test` - кавычки позволяют выполнять запуск из файла README.md). Для оформления данных в виде таблице используй сочетание клавиш Ctrl + N или Cmd + N. Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

SergeyDyomin92 committed:
Diplom-3 task
 Исправлено замечание: Для всех полей указаны минимально необходимые модификаторы доступа.
 Добавлено улучшение: Заполнен файл README.md. Тесты перезапущены и сгенерирован новый отчет.
